pkg/zng/docs/spec.md

```diff
@@ -1,92 +1,366 @@
-# ZSON specification
+# ZNG Specification
 
-ZSON is a format for structured data values, ideally suited for streams
-of heterogeneously typed records.
-ZSON is richly typed and thinner than JSON.
+ZNG is a format for structured data values, ideally suited for streams
+of heterogeneously typed records.  ZNG has both a text form simply called "ZNG",
+comprised of a sequence of newline-delimited UTF-8 strings,
+as well as a binary form called "BZNG", which is based on machine-readable data types
+inspired by [Protobufs](https://developers.google.com/protocol-buffers).
+
+ZNG is richly typed and thinner on the wire than JSON.
 Like [newline-delimited JSON (NDJSON)](http://ndjson.org/),
-ZSON represents a sequence of data objects that can be parsed line by line.
+the ZNG text format represents a sequence of data objects
+that can be parsed line by line.
 
-ZSON strikes a balance between the narrowly typed but flexible NDJSON format and
+ZNG strikes a balance between the narrowly typed but flexible NDJSON format and
 a more structured approach like
 [Apache Avro](https://avro.apache.org).
-ZSON is type rich and
-embeds all type/schema in the stream, while having a value syntax
-independent of the schema, making it easy and efficient to parse on the fly
-and mix and match streams from different sources with heterogeneous types.
-Like Avro,
-ZSON embeds schema information in the data stream but ZSON schema bindings
-are lighter-weight and specified with a simple integer encoding that
-accompanies each data value.
-
-The ZSON design [is also motivated by](./rationale.md)
+ZNG is type rich and embeds all type/schema in the stream while having a
+binary serialization format that allows "lazy parsing" of fields such that
+only the fields of interest in a stream need to be extracted and parsed.
+Unlike Avro and Protobufs,
+ZNG embeds type information in the data stream and admits efficient
+multiplexing of heterogenous data types by including with each data value
+a simple integer identifier to reference its type.
+
+ZNG is a superset of JSON in that any JSON input
+can be mapped onto ZNG and recovered by decoding
+that ZNG back into JSON.
+
+The ZNG design [is also motivated by](./rationale.md)
 and maintains backward compatibility with the original [default Zeek log format](https://docs.zeek.org/en/stable/examples/logs/).
```

Comment:
Is this still technically true given the "WE DECIDED TO FOREGO COMPLETE FORMAT-COMPATIBILITY BETWEEN ZNG AND ZEEK" over at [zeek-compat.md](https://github.com/mccanne/zq/blob/zng-spec/pkg/zng/docs/zeek-compat.md)?

Reply:
Maybe "compatibility" instead of "backward compatibility"

Reply:
fixed